controllers/front/ValidateOrder.php

```diff
@@ -241,7 +246,7 @@ private function handleException(Exception $exception)
                 $this->redirectToCheckout(['step' => 'payment', 'paymentError' => $exception->getCode()]);
                 break;
             case PayPalException::ORDER_ALREADY_CAPTURED:
-                $this->module->currentOrder = (new \OrderMatrice())->getOrderPrestashopFromPaypal(Tools::getValue('orderId'));
+                $this->module->currentOrder = (new \OrderMatrice())->getOrderPrestashopFromPaypal($paypalOrder);
```

Comment:
I haven't seen the specific case 'invalid' to be handled here or in `getOrderPrestashopFromPaypal`. This means we might find a PrestaShop Order and redirect the customer to a wrong OrderConfirmation, don't you think ?

Reply:
Cannot be happens, PsCheckoutException::PAYPAL_ORDER_IDENTIFIER_MISSING will be throw before ;)